pkg/schedulingtypes/plugin.go

```diff
@@ -133,6 +143,24 @@ func (p *Plugin) FederatedTypeExists(key string) bool {
 	return exist
 }
 
+func (p *Plugin) GetResourceClusters(qualifiedName util.QualifiedName, clusters []*fedv1b1.KubeFedCluster) (selectedClusters sets.String, err error) {
+	fedObject, err := p.federatedTypeClient.Resources(qualifiedName.Namespace).Get(context.Background(), qualifiedName.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// get FederatedNamespace with namespace name of the object
+	fedNsObject, err := p.fedNsClient.Resources(qualifiedName.Namespace).Get(context.Background(), qualifiedName.Namespace, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if p.typeConfig.GetNamespaced() {
+		return util.ComputeNamespacedPlacement(fedObject, fedNsObject, clusters, p.limitedScope)
+	}
+	return util.ComputePlacement(fedObject, clusters)
```

Comment:
Please see my comment above about using this as is here. I think we need a separate function which gets only the clusterselector from the target resource. We are trying to intersect with the clusterselector only, not the clusternames.